### PR TITLE
MOSIP-23954 MOSIP-24002 MOSIP-24286 subtypes for biometric capture, Integration with authFactors, Language fallback implementation, login with OTP UI changes

### PR DESCRIPTION
**Pull request by anshulv1401**
[MOSIP-23954](https://mosip.atlassian.net/browse/MOSIP-23954)
[MOSIP-24002](https://mosip.atlassian.net/browse/MOSIP-24002)
[MOSIP-24286](https://mosip.atlassian.net/browse/MOSIP-24286)
[MOSIP-24665](https://mosip.atlassian.net/browse/MOSIP-24665)

1. subtype configuration added for biometric capture request.
2. i18n support added for claims.
3. if idpApiUrl starts with "/" then baseUrl will be the concatenation of origin-domain and idpApiUrl.
4. transactionId for bio captured reverted to idp transactionId.
5. relative api-url added in the configuration file.
6. Integration with authFactors.
7. Landing page becomes the 1st object in auth factor.
8. More ways are shown only if more than 1 object exists. will show the list of all auth factors, ignoring navigation for multi factors for now.
9. Language preference: ui_locales parameter
10. Language preference: cookie
11. Language preference: browser's setting
12. Language preference: Default language from the properties file.
13. Added default IDP details page
14. LoginWithOTP UI updated and integrated with IDP APIs.
15. Issue (MOSIP-24665) Dynamically load i18n file fixed

### CODE REVIEW

File: src/components/IdpDetails.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const { t } = useTranslation("idpDetails");
+
+  const [status, setStatus] = useState({ state: states.LOADED, msg: "" });
+  const [details, setDetails] = useState(null);
+
+  useEffect(() => {
+    setStatus({ state: states.LOADING, msg: t("loading_msg") });
+
+    let detailList = [
+      { icon: null, name: "internal_api", value: "https://api.dev.mosip.net" },
```
Comment (gsasikumar) [inline src/components/IdpDetails.js:15]: Why is this hard coded?

File: src/constants/clientConstants.js [hunk 2]
```diff
@@ -24,6 +25,15 @@ const configurationKeys = {
   sbiFaceCaptureScore: "sbi.capture.score.face",
   sbiFingerCaptureScore: "sbi.capture.score.finger",
   sbiIrisCaptureScore: "sbi.capture.score.iris",
+
+  sbiIrisBioSubtypes: "sbi.bio.subtypes.iris", //comma separated list of bio-subtypes. default is "UNKNOWN"
+  sbiFingerBioSubtypes: "sbi.bio.subtypes.finger", //comma separated list of bio-subtypes. default is "UNKNOWN"
+};
+
+const supportedLanguages = {
```
Comment (gsasikumar) [inline src/constants/clientConstants.js:33]: this should be configurable

File: src/i18n.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import i18n from "i18next";
 import { initReactI18next } from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
 
 // Importing translation files
 
 import translationEN from "./locales/en.json";
```
Comment (gsasikumar) [inline src/i18n.js:7]: All these files should be dynamically picked.

File: src/i18n.js [hunk 4]
```diff
@@ -18,31 +19,37 @@ const resources = {
...
-  hn: {
-    consent: translationHE.consent,
-    l1Biometrics: translationHE.l1Biometrics,
-    pin: translationHE.pin,
-    signInOption: translationHE.signInOption,
-    header: translationHE.header,
-    authorize: translationHE.authorize,
-    tabs: translationHE.tabs,
-    errors: translationHE.errors,
+  hi: {
```
Comment (gsasikumar) [inline src/i18n.js:24]: whats this? configuration per language?

File: src/locales/en.json [hunk 5]
```diff
@@ -5,13 +5,32 @@
   },
   "consent": {
     "authorization_failed_msg": "Authorization failed!",
+    "consent_request_rejected": "Consent Request Rejected!",
```
Comment (gsasikumar) [inline src/locales/en.json:8]: All the locales have to come from artifactory url. Can you speak to mayura or aravindan once to understand how they are getting this file?

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: public/env_configs.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
 window["envConfigs"] = {
-  idpApiUrl: "https://idp.dev.mosip.net/v1/idp",
+  idpApiUrl: "http://localhost:8088/v1/idp",
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply public/env_configs.js]: idpApiUrl removed from all places. Relative URL "/v1/idp" added in env file, which will be constant.

File: src/constants/clientConstants.js (hunk 2 above)
Comment (anshulv1401) [reply src/constants/clientConstants.js:33]: these are configuration keys, values for which will be sent from the IDP OAuth-details endpoint.

File: src/i18n.js (hunk 3 above)
Comment (anshulv1401) [reply src/i18n.js:5]: Raised a ticked to dynamically load i18n file
[MOSIP-24665](https://mosip.atlassian.net/browse/MOSIP-24665)

File: src/locales/en.json (hunk 5 above)
Comment (anshulv1401) [reply src/locales/en.json:8]: Raised a ticked to download i18n bundle from artifactory and host from idp-ui nginx
[MOSIP-24666](https://mosip.atlassian.net/browse/MOSIP-24666)

File: src/i18n.js (hunk 4 above)
Comment (anshulv1401) [reply src/i18n.js]: removed

File: src/i18n.js (hunk 3 above)
Comment (anshulv1401) [reply src/i18n.js]: issue fixed

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,27 +1,58 @@
...
+1. Authorize
+2. Login
+3. Consent
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+* /Authorize:
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply README.md]: changed it to /authorize

File: README.md [hunk 8]
```diff
@@ -1,27 +1,58 @@
...
+    /Authorize a redirect URI which needs to be called from the relying party portal. This is the endpoint that handles the oidc authorization. For API details, refer to this [document](https://mosip.stoplight.io/docs/identity-provider/85f761d237115-authorization-endpoint). 
 
-`npm test`
+* /Login:
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply README.md]: changed it to /login

File: README.md [hunk 9]
```diff
@@ -1,27 +1,58 @@
...
+    Login is a page that has the following components.
+    -   Sign in with Biometrics
+    -   Sign in with OTP
+    -   Sign in with PIN(MOCK)
+    -   Sign in with Inji(impl in progress)
 
-Launches the test runner in the interactive watch mode.\
+    Each component is capable of authorizing the user using its challenge details. These options to sign in will be presented to the user based on the auth factors that get returned from the /Authorize endpoint.
 
+* /Consent: Consent is a page that prompts the user to provide consent to share one's details from the MOSIP to the relying party. It shows authorize scope that needs to be permitted and essential and voluntary claims that need to be accepted or rejected.
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply README.md]: /consent

File: README.md [hunk 10]
```diff
@@ -1,27 +1,58 @@
...
-`npm test`
+* /Login:
+    Login is a page that has the following components.
+    -   Sign in with Biometrics
+    -   Sign in with OTP
+    -   Sign in with PIN(MOCK)
+    -   Sign in with Inji(impl in progress)
 
-Launches the test runner in the interactive watch mode.\
+    Each component is capable of authorizing the user using its challenge details. These options to sign in will be presented to the user based on the auth factors that get returned from the /Authorize endpoint.
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply README.md]: Paragraph modified

File: README.md [hunk 11]
```diff
@@ -1,27 +1,58 @@
-# Introduction 
-[EXPERIMENTAL]
-The app is the user interface for the mosip idp. 
+# IDP UI
 
-### Getting Started
+## Overview
+This repository contains the source code for the UI of MOSIP IDP Services.
+IDP Services are based on [OpenID specs](https://openid.net/specs/openid-connect-core-1_0.html).
```
[the remark this answers is not in the document]
Comment (ase-101) [reply README.md:5]: we will take care of this in https://mosip.atlassian.net/browse/MOSIP-24791

**Review by gsasikumar — APPROVED** [review]

**Pull request merged**